tests/get_socket_test.py

```diff
@@ -105,9 +105,8 @@ def test_get_socket_os_error():
     connection_manager = adafruit_connection_manager.ConnectionManager(mock_pool)
 
     # try to get a socket that returns a OSError
-    with pytest.raises(RuntimeError) as context:
+    with pytest.raises(OSError):
         connection_manager.get_socket(mocket.MOCK_HOST_1, 80, "http:")
-    assert "Error connecting socket: OSError" in str(context)
```

Comment:
Why did these get removed? I had these so we make sure that an accidental code change doesn't break error handling. This one did (which is fine), so ideally we would update the assert.

Reply:
Some of these are now direct exceptions, instead of exceptions wrapping up a previous exception. So they have no message.

Reply:
Interesting. They should still have a message. I'll follow up with a testing PR once this is merged.

Reply:
I could not see one in some cases. Let me know.

Reply:
@justmobilize 
> I'll follow up with a testing PR once this is merged.

So it's OK with you to merge now? If you fix the testing here to be more precise, does that affect the MiniMQTT and Requests PR's. If not I will merge those after they pass.

Reply:
@dhalbert, yes this is good to merge.

My updates will just be to the testing files